Core/Src/main.c

```diff
@@ -93,18 +95,20 @@ int main(void)
   MX_SPI1_Init();
   MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-  HAL_TIM_PWM_Init(&htim1);
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-	  HAL_SPI_TransmitReceive_IT(&hspi1, &tx_buf, rx_buf, sizeof(uint8_t));
-	  *rx_buf = (uint8_t)(65535/20) + *rx_buf/20; //scales ADC digital value to be between 5-10% of 65535
-	  *rx_buf = *rx_buf & 65528; // since max value of scaled ADC value has 13 bits, we can keep most significant 10 bits
-	  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, *rx_buf); //set compare register
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	  HAL_SPI_TransmitReceive_IT(&hspi1, &tx_buf, rx_buf, sizeof(rx_buf));
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+	  *rx_buf = (uint8_t)(0.05 * MAX_TIMER_VALUE) + 0.05 * (*rx_buf) * (MAX_ADC_VALUE/MAX_TIMER_VALUE); //scales ADC digital value to be between 5-10% of 65535
```

Comment:
This line is still confusing and I dislike anything writing to the rx_buf other than in the transmit and receive function.
What I see it doing here is:
casting 5% of the MAX_TIMER_VALUE to a byte, adding it to 5% of the received ADC value times the MAX_ADC_VALUE / MAX_TIMER_VALUE.
I highly recommend adding a MIN_TIMER_VALUE constant, and converting the ADC value to a percent or decimal on a different line.
I also just noticed that though you only care about 10 of the 16 bits you read back, the other 6 can still affect the ADC returned value. You will need to extract just the 10 bits of actual data, I recommend having a look at bit masking for this.

Reply:
Not sure about the MIN_TIMER_VALUE but I converted the ADC value to be out of 64000 on a separate line and discarded the 6 MSB.